infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -374,4 +464,41 @@ public void shutdown() {
     public void enforceSnapshotSync(DiscoveryServiceEvent event) {
         replicationManager.enforceSnapshotSync(event);
     }
+
+    /**
+     * Check if the local cluster is a connection receiver for any of the discovered sessions
+     *
+     * @return true if the local cluster is a connection receiver for any session
+     */
+    public boolean isConnectionReceiver() {
+        Set<String> connectionEndpoints = topology.getRemoteClusterEndpoints().keySet();
+        for(LogReplicationSession session : sessions) {
+            if (!connectionEndpoints.contains(session.getSinkClusterId()) &&
+                    !connectionEndpoints.contains(session.getSourceClusterId())) {
+                return true;
```

Comment:
true should be returned if connection endpoints does not contain the source or sink cluster id for any session, not just 1, right?

Or this can be changed to 'or' and return false if any session has a cluster id in the endpoints.

Reply:
Here we are just checking if the local cluster is a connection receiver for any session.
Note, a cluster can be both connection receiver and connection initiator at the same time.

We start/stop the interClusterServerNode depending on this method's return value.
If the local cluster is not a connection receiver, then we don't need to have the interClusterServerNode started.